test/unit_tests/arkode/CXX_serial/ark_test_butcher.cpp

```diff
@@ -21,76 +21,43 @@
-                                                   "ARKODE_ARK436L2SA_DIRK_6_3_4",
-                                                   "ARKODE_KVAERNO_7_4_5",
-                                                   "ARKODE_ARK548L2SA_DIRK_8_4_5",
-                                                   "ARKODE_ARK437L2SA_DIRK_7_3_4",
-                                                   "ARKODE_ARK548L2SAb_DIRK_8_4_5",
-                                                   "ARKODE_ESDIRK324L2SA_4_2_3",
-                                                   "ARKODE_ESDIRK325L2SA_5_2_3",
-                                                   "ARKODE_ESDIRK32I5L2SA_5_2_3",
-                                                   "ARKODE_ESDIRK436L2SA_6_3_4",
-                                                   "ARKODE_ESDIRK43I6L2SA_6_3_4",
-                                                   "ARKODE_QESDIRK436L2SA_6_3_4",
-                                                   "ARKODE_ESDIRK437L2SA_7_3_4",
-                                                   "ARKODE_ESDIRK547L2SA_7_4_5",
-                                                   "ARKODE_ESDIRK547L2SA2_7_4_5"};
-  std::vector<ARKODE_ERKTableID> Tables_ARK_ERK = {ARKODE_ARK2_ERK_3_1_2,
-                                                   ARKODE_ARK324L2SA_ERK_4_2_3,
-                                                   ARKODE_ARK436L2SA_ERK_6_3_4,
-                                                   ARKODE_ARK437L2SA_ERK_7_3_4,
-                                                   ARKODE_ARK548L2SA_ERK_8_4_5,
-                                                   ARKODE_ARK548L2SAb_ERK_8_4_5};
-  std::vector<ARKODE_DIRKTableID> Tables_ARK_DIRK =
-    {ARKODE_ARK2_DIRK_3_1_2,       ARKODE_ARK324L2SA_DIRK_4_2_3,
-     ARKODE_ARK436L2SA_DIRK_6_3_4, ARKODE_ARK437L2SA_DIRK_7_3_4,
-     ARKODE_ARK548L2SA_DIRK_8_4_5, ARKODE_ARK548L2SAb_DIRK_8_4_5};
-  std::vector<std::string> STables_ARK = {"ARKODE_ARK2_3_1_2",
-                                          "ARKODE_ARK324L2SA_4_2_3",
-                                          "ARKODE_ARK436L2SA_6_3_4",
-                                          "ARKODE_ARK437L2SA_7_3_4",
-                                          "ARKODE_ARK548L2SA_8_4_5",
-                                          "ARKODE_ARK548L2SAb_8_4_5"};
-  int numfails                         = 0;
+  std::vector<ARK_Table> ark_tables =
+    {{"ARKODE_ARK2_3_1_2", "ARKODE_ARK2_ERK_3_1_2", "ARKODE_ARK2_DIRK_3_1_2"},
+     {"ARKODE_ARK324L2SA_4_2_3", "ARKODE_ARK324L2SA_ERK_4_2_3",
+      "ARKODE_ARK324L2SA_DIRK_4_2_3"},
+     {"ARKODE_ARK436L2SA_6_3_4", "ARKODE_ARK436L2SA_ERK_6_3_4",
+      "ARKODE_ARK436L2SA_DIRK_6_3_4"},
+     {"ARKODE_ARK437L2SA_7_3_4", "ARKODE_ARK437L2SA_ERK_7_3_4",
+      "ARKODE_ARK437L2SA_DIRK_7_3_4"},
+     {"ARKODE_ARK548L2SA_8_4_5", "ARKODE_ARK548L2SA_ERK_8_4_5",
+      "ARKODE_ARK548L2SA_DIRK_8_4_5"},
+     {"ARKODE_ARK548L2SAb_8_4_5", "ARKODE_ARK548L2SAb_ERK_8_4_5",
+      "ARKODE_ARK548L2SAb_DIRK_8_4_5"}};
```

Comment:
This piece is no more easily "maintained" than what was there before, as any new ARK table will still require manual intervention to add it here.

Reply:
Yeah, there not much that we can do for ARK methods. That said, I don't think this should discourage us in the ERK/DIRK case.